### PR TITLE
Fix mock test for AlloyDB Cluster

### PR DESCRIPTION
**Pull request by anhdle-sso**
### Change description

- Fix String function for Mock AlloyDB Cluster
- Fix GetCluster function for the mock
- Fix the incorrect API group for alloydb
- Add normalizer for http log
- Add golden file output and its normalizer

### Tests you have done

- [x] Run `make ready-pr` to ensure this PR is ready for review.
- [x] Perform necessary E2E testing for changed resources.

### CODE REVIEW

**Comment by justinsb** [comment]
Looks good.  I think this would be caught by hack/record-gcp / hack/compare-mock, we just aren't running them here?

I think this is in your other PR also, so I'll hold but you can remove hold as you see fit (I would guess the other one, otherwise you'll have to rebase that one) 

/approve
/lgtm
/hold

**Comment by anhdle-sso** [comment]
> Looks good. I think this would be caught by hack/record-gcp / hack/compare-mock, we just aren't running them here?
> 
> I think this is in your other PR also, so I'll hold but you can remove hold as you see fit (I would guess the other one, otherwise you'll have to rebase that one)
> 
> /approve /lgtm /hold

There is a few blunders in the previous PR. The API group was incorrect so the mock test just skipped and the test showed as passed. For the record_gcp, I still have good amount of flakes due to duplicate GET requests. I will debug that issue around the duplicate GET requests in a separate thread. Also, I add the golden file and its normalizer; which was not the norm previously.

File: config/tests/samples/create/harness.go [hunk 1]
```diff
@@ -542,7 +542,7 @@ func MaybeSkip(t *testing.T, name string, resources []*unstructured.Unstructured
 			}
 
 			switch gvk.GroupKind() {
-			case schema.GroupKind{Group: "alloydb.googleapis.com", Kind: "AlloyDBCluster"}:
+			case schema.GroupKind{Group: "alloydb.cnrm.cloud.google.com", Kind: "AlloyDBCluster"}:
```
Comment (justinsb) [inline config/tests/samples/create/harness.go:545]: lol - sorry I missed that!

File: tests/e2e/unified_test.go [hunk 2]
```diff
@@ -563,6 +582,9 @@ func testFixturesInSeries(ctx context.Context, t *testing.T, testPause bool, can
 					for k := range operationIDs {
 						normalizers = append(normalizers, ReplaceString(k, "${operationID}"))
 					}
+					for k := range networkIDs {
```
Comment (justinsb) [inline tests/e2e/unified_test.go:585]: Might be easier to put this in as a pathID, but not a big deal!

**Pull request merged**